broker/lib/middleware.js

```diff
@@ -21,6 +25,69 @@ exports.isFeatureEnabled = function (featureName) {
   };
 };
 
+exports.validateRequest = function (operationType) {
+  return function (req, res, next) {
+    /* jshint unused:false */
+    if (req.instance.async && (_.get(req, 'query.accepts_incomplete', 'false') !== 'true')) {
+      next(new UnprocessableEntity('This request requires client support for asynchronous service operations.', 'AsyncRequired'));
+    }
+    if (_.includes([CONST.OPERATION_TYPE.CREATE], operationType) &&
+      (!_.get(req.body, 'space_guid') || !_.get(req.body, 'organization_guid'))) {
+      next(new BadRequest('This request is missing mandatory organization guid and/or space guid.'));
+    }
+    next();
+  };
+};
+
+exports.lock = function (operationType, lastOperationCall) {
+  return function (req, res, next) {
+    if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR && config.enable_service_fabrik_v2) {
+      // Acquire lock for this instance
+      return lockManager.lock(req.params.instance_id, {
+          lockedResourceDetails: {
+            resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+            resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+            resourceId: req.params.instance_id,
+            operation: operationType ? operationType : req.query.operation.type // This is for the last operation call
+          }
+        })
+        .then(() => next())
+        .catch((err) => {
+          logger.debug('[LOCK]: exception occurred; Need not worry as lock is probably set --', err);
+          //For last operation call, we ensure migration of locks through this
+          if (lastOperationCall && err instanceof DeploymentAlreadyLocked) {
+            logger.info(`Proceeding as lock is already acquired for the resource: ${req.params.instance_id}`);
+            next();
+          } else {
+            next(err);
+          }
+        });
+    }
+    next();
+  };
+};
+
+exports.isWriteLocked = function () {
```

Comment:
can name this method to `checkBlockingOperationInProgress`  or `checkWriteLocked` -- as this is really not returning a boolean.